### PR TITLE
[plugin.video.reddit_viewer] 3.2.3

### PR DESCRIPTION
**Pull request by gedisony**
### Description

Reorganize program structure
Fix bugs

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.reddit_viewer/resources/lib/utils.py [hunk 1]
```diff
@@ -0,0 +1,983 @@
...
+    osAndroid = xbmc.getCondVisibility('System.Platform.Android')
+    #url = 'http://www.google.com/'
+
+    import xbmcaddon
+    addon=xbmcaddon.Addon()
+    custom_link_command=addon.getSetting('custom_link_command')
+    if custom_link_command:
+        custom_link_command=custom_link_command.replace('{url}',url)
+        log('Running custom command for link:\n' + custom_link_command)
+        exec( custom_link_command )
```
Comment (gedisony) [inline plugin.video.reddit_viewer/resources/lib/utils.py]: This allows the user to customize which web browser to follow a link posted on reddit.

File: plugin.video.reddit_viewer/addon.xml [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
-        <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
-        <email>gediso+ny at gmail dot com</email>
-		
-    </extension>
+	id="plugin.video.reddit_viewer"
+	name="Reddit Viewer"
+	version="3.2.3"
+	provider-name="gedisony">
+	<requires>
+		<import addon="xbmc.python"                     version="2.14.0"                />
```
Comment (enen92) [inline plugin.video.reddit_viewer/addon.xml]: addon api version is `2.24.0` for Jarvis

File: plugin.video.reddit_viewer/addon.xml [hunk 3]
```diff
@@ -1,33 +1,49 @@
...
+		<import addon="xbmc.addon"                      version="13.0.0"                />
```
Comment (enen92) [inline plugin.video.reddit_viewer/addon.xml]: this is not needed

File: plugin.video.reddit_viewer/default.py [hunk 4]
```diff
@@ -1270,480 +669,291 @@ def listVideos(url, name, subreddit):
...
+        post_title=n+title+"[CR]"+title_line2
+    else:
+        post_title=n+title
+        il_description=title_line2+"[CR]"+il_description
+
+    il={"title": post_title, "plot": il_description, "plotoutline": il_description, "Aired": credate, "mpaa": mpaa, "Genre": "r/"+subreddit, "studio": domain, "director": posted_by }   #, "duration": 1271}   (duration uses seconds for titan skin
+
+    liz=xbmcgui.ListItem(label=post_title,
+                          label2="",
+                          iconImage="",
```
Comment (enen92) [inline plugin.video.reddit_viewer/default.py]: iconImage and thumbnailImage are both deprecated use liz.setArt instead with a dictionary of image resources

File: plugin.video.reddit_viewer/default.py [hunk 5]
```diff
@@ -1852,531 +1054,396 @@ def playYTDLVideo(url, name, type):
...
+#                                 label2="",
+#                                 iconImage="",
+#                                 thumbnailImage='',
+#                                 path=DirectoryItem_url)
+
+            liz=xbmcgui.ListItem(label=list_title +': '+ desc100 ,
                                  label2="",
-                                 iconImage="DefaultVideo.png", 
-                                 thumbnailImage=thumb_url)
+                                 iconImage="",
```
Comment (enen92) [inline plugin.video.reddit_viewer/default.py]: same

File: plugin.video.reddit_viewer/default.py [hunk 6]
```diff
@@ -1852,531 +1054,396 @@ def playYTDLVideo(url, name, type):
...
+                                 thumbnailImage="")
 
-            
-            liz.setInfo( type="Video", infoLabels={ "Title": h[1], "plot": result, "studio": hoster } )
-            liz.setArt({"thumb": thumb_url, "poster":thumb_url, "banner":thumb_url, "fanart":thumb_url, "landscape":thumb_url   })
+            liz.setInfo( type="Video", infoLabels={ "Title": h[1], "plot": plot, "studio": domain, "votes": str(comment_score), "director": author  } )
```
Comment (enen92) [inline plugin.video.reddit_viewer/default.py]: this was correct..

File: plugin.video.reddit_viewer/default.py [hunk 7]
```diff
@@ -1852,531 +1054,396 @@ def playYTDLVideo(url, name, type):
...
+                directory_items.append( (DirectoryItem_url, liz, isFolder,) )
+                #xbmcplugin.addDirectoryItem(handle=pluginhandle,url=DirectoryItem_url,listitem=liz,isFolder=isFolder)
+            else:
+                #this section are for comments that have no links
+                if not ShowOnlyCommentsWithlink:
+                    result=h[3].replace('](', '] (')
+                    result=markdown_to_bbcode(result)
+                    liz=xbmcgui.ListItem(label=list_title + desc100 ,
+                                         label2="",
+                                         iconImage="",
```
Comment (enen92) [inline plugin.video.reddit_viewer/default.py]: drop those

File: plugin.video.reddit_viewer/resources/lib/guis.py [hunk 8]
```diff
@@ -0,0 +1,558 @@
...
+        # it might be possible to make the textbox control in code and addcontrol() it. but then you would have to figure out how to get text to change when listbox selection changes.
+        #if self.plot_font:
+        #    self.ctl_plot_textbox = self.getControl(self.CONTROL_ID_FOR_PLOT_TEXTBOX)
+        #    self.ctl_plot_textbox.setLabel('Status', 'font14', '0xFFFFFFFF', '0xFFFF3300', '0xFF000000')
+
+        #url="plugin://plugin.video.reddit_viewer/?url=plugin%3A%2F%2Fplugin.video.youtube%2Fplay%2F%3Fvideo_id%3D73lsIXzBar0&mode=playVideo"
+        #url="http://i.imgur.com/ARdeL4F.mp4"
+        if self.include_parent_directory_entry:
+            back_image='DefaultFolderBackSquare.png'
+            listitem = xbmcgui.ListItem(label='..', label2="", iconImage=back_image)
```
Comment (enen92) [inline plugin.video.reddit_viewer/resources/lib/guis.py]: drop the iconImage and use it in setArt

**Comment by enen92** [comment]
Please adjust my comments. Two things to keep in mind for the future:
* Please consider to remove commented code, it just increases the plugin size
* The default.py file is too big and is the main file of your addon. Usually, the entry point for your plugin should be the smallest possible. It should be used just to import the main file and run it. Keep in mind python won't compile this file which may reduce your plugin performance. See how it is done on the addon-generator: 
The main file (defined in addon.xml) is minimal and is only used to invoke the main plugin python file: https://github.com/xbmc/generator-kodi-addon/blob/master/generators/app/templates/main.py

**Comment by gedisony** [comment]
Thanks for taking the time to review my addon. 
I've removed the comments and made the requested changes. 
Keeping the main addon file small will need more testing. 
Please consider this version while I work on moving the functions out of default.py for the next release.

**Comment by enen92** [comment]
Latest comment was just something to keep in mind :)

**Pull request merged**